src/NServiceBus.RabbitMQ/RabbitMQTransportInfrastructure.cs

```diff
@@ -25,13 +25,24 @@ public RabbitMQTransportInfrastructure(SettingsHolder settings, string connectio
             var connectionConfiguration = new ConnectionStringParser(settings).Parse(connectionString);
             connectionFactory = new ConnectionFactory(connectionConfiguration);
 
+            SetUsePublisherConfirmsIfPresent(settings, connectionConfiguration);
+
             CreateTopology();
 
             channelProvider = new ChannelProvider(connectionFactory, routingTopology, connectionConfiguration.UsePublisherConfirms);
 
             RequireOutboxConsent = false;
         }
 
+        static void SetUsePublisherConfirmsIfPresent(SettingsHolder settings, ConnectionConfiguration connectionConfiguration)
+        {
+            if (settings.HasSetting(SettingsKeys.UsePublisherConfirms))
+            {
+                var publisherCofirms = settings.Get<bool>(SettingsKeys.UsePublisherConfirms);
+                connectionConfiguration.UsePublisherConfirms = publisherCofirms;
+            }
```

Comment:
@timbussmann can you help on this. I noticed that we have:
SettingsHolder that contains Default values and overrides. Thanks to that you can then use
settings.GetOrDefault that returns default if the override is not present.

However in Rabbit we set the defaults not in the Settings but in here:
https://github.com/Particular/NServiceBus.RabbitMQ/blob/5f0164ca89c828c860b49ba1bc44c9b03dda93be/src/NServiceBus.RabbitMQ/Configuration/ConnectionConfiguration.cs#L36-L51

Should we set Defaults on settings? Or was it created for a different purpose? That would simplify the code above